### PR TITLE
refactor: no executor tests

### PR DESCRIPTION
**Pull request by florian-hoenicke**
Both jobs had the same precondition and therefore can be joined.
The executor tests are benefiting from all the runners we have since they are now distributed.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -275,46 +275,6 @@ jobs:
...
-      #        timeout-minutes: 60
-      - name: Upload coverage
-        uses: actions/upload-artifact@v2
-        with:
-          name: coverage_executor
-          path: .coverage
-
   # just for blocking the merge until all parallel core-test are successful
   success-all-test-coverage:
     runs-on: ubuntu-latest
```
Comment (violenil) [inline .github/workflows/ci.yml:280]: I think this 'needs' also should be updated to remove `executor-test`

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -53,36 +53,6 @@ jobs:
...
-          pytest --suppress-no-test-exit-code -v -s tests/executor/indexer/elastic
-          pytest --suppress-no-test-exit-code -v -s tests/executor/preprocessor
-#      - name: Setup tmate session for debugging
-#        if: ${{ failure() || (github.event_name == 'workflow_dispatch' && github.event.inputs.debug_enabled) }}
-#        uses: mxschmitt/action-tmate@v3
-#        timeout-minutes: 60
-
 
   prerelease:
     runs-on: ubuntu-latest
```
Comment (violenil) [inline .github/workflows/cd.yml:58]: same here

File: .github/workflows/ci.yml (hunk 1 above)
Comment (florian-hoenicke) [reply .github/workflows/ci.yml:280]: thanks - done

File: .github/workflows/cd.yml (hunk 2 above)
Comment (florian-hoenicke) [reply .github/workflows/cd.yml:58]: thanks - done

**Comment by florian-hoenicke** [comment]
test if it fixed by the core

**Pull request merged**